source/plugins/go/src/log_loss_monitoring.go

```diff
@@ -0,0 +1,277 @@
+containerIds back to a second shared channel (deleted_containers_response)
+
+4. The Main thread will read from deleted_containers_response (also non-blocking) and delete any recieved
+containerIDs from it's own tables. (the delay is so that any logs from a freshly deleted container will have time to go through fbit)
+
+Motivation: with this method, the Main thread should store about 1.2x as many containers as are on disk (assuming it queries about deleted containers
+in a evenly distributed way). If more than 1/5 of the containers in the Main thread's data structures are deleted, then on average query more
+than one deleted containerId. That will cause the number of containers the Main thread is storing to shrink.
+*/
+
+// Main thread managed data structures:
+// var m_container_to_arr_index = make(map[string]int64)
+var m_bytes_logged_storage AddressableMap = Make_AddressableMap()
+
+var m_deletion_query_index = 0
+
+// File Watcher managed data structures
+// (FW = File Watcher)
+type FwRecord struct {
+	unrotated_bytes    int64
+	rotated_bytes      int64
+	existing_log_files []string
+}
+
+var FW_records = make(map[string]FwRecord) // identifier -> file names
+// var FW_bytes_logged_rotated = make(map[string]int64)   // identifier -> count
+// var FW_bytes_logged_unrotated = make(map[string]int64) // identifier -> count
+// var current_iteration_count int64 = 0
+var num_containers_on_disk int32 = 0
+
+// Shared objects
+var deleted_containers_query = make(chan string, 5)    // There might be a lot more than 5 containers  of 10 chosen arbitrairly
+var deleted_containers_response = make(chan string, 5) // capacity of 10 chosen arbitrairly
+
+var process_logs_mut = &sync.Mutex{}
+var read_disk_mut = &sync.Mutex{}
+var enabled bool
+
+var disabled_namespaces = make(map[string]bool)
+
+var log_loss_logger *log.Logger
+
+func SetupLogLossTracker() {
+	enabled = os.Getenv("CONTROLLER_TYPE") == "DaemonSet"
+
+	enabled = enabled && !(strings.ToLower(os.Getenv("IN_UNIT_TEST")) == "true")
+
+	// toggle env var is meant to be set by Microsoft, customer can set AZMON_DISABLE_LOG_LOSS_TRACKING through a configmap
+	enabled = enabled && (os.Getenv("AZMON_DISABLE_LOG_LOSS_TRACKING") != "true") && (os.Getenv("AZMON_DISABLE_LOG_LOSS_TRACKING_TOGGLE") != "true")
+
+	// don't count logs if stdout or stderr log collection is globally disabled
+	enabled = enabled && strings.ToLower(os.Getenv("AZMON_COLLECT_STDOUT_LOGS")) == "true" && strings.ToLower(os.Getenv("AZMON_COLLECT_STDERR_LOGS")) == "true"
+
+	if enabled {
+		log_loss_logger = createLogger("container-log-counts.log")
+
+		for _, excluded_namespace := range strings.Split(os.Getenv("AZMON_STDERR_EXCLUDED_NAMESPACES"), ",") {
+			disabled_namespaces[excluded_namespace] = true
+		}
+
+		for _, excluded_namespace := range strings.Split(os.Getenv("AZMON_STDOUT_EXCLUDED_NAMESPACES"), ",") {
+			disabled_namespaces[excluded_namespace] = true
```

Comment:
What happens when a namespace is excluded for only stdout or stderr but not both?

Reply:
I'm completely ignoring namespaces with any customization on them (including stdout or stderr disabled), no data is recorded from them